source/modulo_controllers/src/BaseControllerInterface.cpp

```diff
@@ -403,10 +406,15 @@ void BaseControllerInterface::add_outputs() {
               [&](StringPublishers& pub) {
                 pub.first = get_node()->create_publisher<std_msgs::msg::String>(topic, qos_);
                 pub.second = std::make_shared<realtime_tools::RealtimePublisher<std_msgs::msg::String>>(pub.first);
+              },
+              [&](CustomPublishers& pub) {
+                custom_output_configuration_callables_.at(name)(pub, name);
               }},
           publishers);
+    } catch (const std::bad_any_cast& ex) {
```

Comment:
Do you want to do the same catch in `add_inputs`?

Reply:
the thing is in `add_inputs` we call the `custom_input_configuration_callables` lamda which the only thing it does is to `push_back` another lambda to `subscriptions`. It's only that lambda that does `std::any_cast`. That's why my original question was about adding a try catch in that lamda.

Now, it could be that someone does `add_input<TypeA>` and then does `read_input<TypeB>`, at which point `read_input` would throw a `bad_any_cast` that is handled. The only way for the subscription lamda to throw would be if a message of different type made it all the way to it. But I think ROS would already complain on its own if a message of `TypeB` is published onto a topic of a different `TypeA`

Long story short, I think we might be safe to not add something more here. Does this sound right to you?

Reply:
Yeah sounds good :+1: